### ISSUE TITLE
tests stall (again!) with recent git-annex

### ISSUE THREAD

**Issue by yarikoptic** [datalad/datalad#4499]
There was a fight for a while trying to figure out what is going on:

- e.g. the https://github.com/datalad/datalad/pull/4265
- annex issue: https://git-annex.branchable.com/bugs/enableremote_stuck_with_a_recentish_git-annex/

which was "mitigated" by adding a skip of the test which checks that git-annex creates a socket if we do not pass configuration option to use the one we create: https://github.com/datalad/datalad/pull/4288 . And I thought that we are "all good"ish at least for now.  But then @kyleam today has mentioned that our cron jobs are hanging! And indeed looking at maint or master -- we have hangs eg https://travis-ci.org/github/datalad/datalad/jobs/683567878
```shell
$ http_proxy= PATH=$PWD/../tools/coverage-bin:$PATH $NOSE_WRAPPER `which nosetests` $NOSE_OPTS -A "$NOSE_SELECTION_OP($NOSE_SELECTION)" --with-doctest --with-cov --cover-package datalad --logging-level=INFO $TESTS_TO_PERFORM
...............
........................................................S........S......................................................
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#build-times-out-because-no-output-was-received
The build has been terminated
```

so saga continues -- we can't upload recent annex to neurodebian until this issue again is mitigated in tests (skips), fixed up (if we do not start ssh socket where we could/should) or properly troubleshooted. heh heh

meanwhile I will build/upload bleeding edge git-annex release to -devel of neurodebian

**Comment by yarikoptic** [comment]
maint branch testing.
From: https://travis-ci.org/github/datalad/datalad/jobs/684702932 
```
datalad.distribution.tests.test_install.test_install_subds_from_another_remote ... .: clone2(?) [ssh://localhost/tmp/datalad_temp_test_install_subds_from_another_remoteoaxxo88n/clone2 (git)]
```
From: https://travis-ci.org/github/datalad/datalad/jobs/684702934
```
datalad.distribution.tests.test_install.test_install_subds_with_space ... 
```

**Comment by yarikoptic** [comment]
a pointer to prior troubleshooting which ran into `enableremote`: https://github.com/datalad/datalad/pull/4116#issuecomment-582791175

**Comment by yarikoptic** [comment]
I @kyleam 's attempt at docker based ssh testing he also [ran into stall](https://github.com/datalad/datalad/issues/4514#issuecomment-628227686), so I thought to mention another vector which might be worth investigation: location where sockets are created. May be annex creates them (e.g. in `enableremote`) on a file system different from the one we use so somehow it plays a role.

**Comment by kyleam** [comment]
At least some of the hanging we're still experiencing should be addressed by git-annex's 8.20200617-134-gf912f8e5f ([associated bug report][0]).  On a Xenial virtual machine, that commit resolves hanging I see with both the localhost and docker ssh targets, specifically in `test_ria_postclonecfg`, `test_install_subds_from_another_remote`, and `test_install_subds_with_space`.

I triggered a [job][0] with the most recent standalone snapshot (8.20200617-144-gb7a78cbb2).  Sadly it still seems to hang, though it is getting farther along.  All the recent cron jobs with recent annex versions were (consistently, I believe) hanging in `core.distributed.tests.test_clone`.  I've triggered [another job][2] with `-v` to see which particular test is hanging.

[0]: https://git-annex.branchable.com/bugs/annex-ssh-options_dropped_since_8.20200330/
[1]: https://travis-ci.org/github/datalad/datalad/builds/704455847
[2]: https://travis-ci.org/github/datalad/datalad/builds/704464417
